packages/contracts/src/vault/Vault.sol

```diff
@@ -8,37 +8,38 @@ import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import { Math } from "openzeppelin-contracts/contracts/utils/math/Math.sol";
 import { Pausable } from "@openzeppelin/contracts/utils/Pausable.sol";
-import { IErrors } from "../interface/IErrors.sol";
 
-abstract contract CredbullBaseVault is ERC4626, Pausable {
+/**
+ * @title Credbull Vault
+ * @author @pasviegas
+ * @notice The family defining contract, based upon Open Zeppelin's ERC4626 implementation.
+ * @dev Yay, dudes, it's awesome with overrides to
+ */
+abstract contract Vault is ERC4626, Pausable {
     using Math for uint256;
 
+    /// @notice Thrown when attempting to create a Credbull Vault with an invalid Custodian Address.
+    error CredbullVault__InvalidCustodianAddress(address);
+    /// @notice Thrown when attempting to create a Credbull Vault with a non-addressable Asset IERC20.
+    error CredbullVault__InvalidAsset(IERC20);
     error CredbullVault__TransferOutsideEcosystem(address);
     error CredbullVault__InvalidAssetAmount(uint256);
     error CredbullVault__UnsupportedDecimalValue(uint8);
     error CredbullVault__NativeTransferNotAllowed();
 
-    /// @notice Struct for Parameters required to create a base vault
-    struct BaseVaultParams {
+    /// @notice The set of parameters required to create a Credbull Vault instance.
+    struct VaultParameters {
```

Comment:
i prefer Params to Parameters.  not sure the verbosity helps with clarity.  i see args as much as arguments for example.